src/credentials/CHIPCertFromX509.cpp

```diff
@@ -693,5 +857,19 @@ DLL_EXPORT CHIP_ERROR ConvertX509CertToChipCert(const uint8_t * x509Cert, uint32
     return err;
 }
 
+DLL_EXPORT CHIP_ERROR ExtractPubkeyFromX509Cert(const uint8_t * x509Cert, uint32_t x509CertLen, Crypto::P256PublicKey & pubkey)
```

Comment:
Wanted to understand the usecase here? Specifically, why there is a need to extract public key from X509 certificate but there is no need to extract public key from the CHIP certificate format?

Reply:
This has been removed as it's more appropriate for a separate PR. But for the usecase: during Device Attestation, we need to extract a public key from a X509 Certificate in order to validate the nonce.